src/Akeneo/Pim/Automation/DataQualityInsights/tests/back/Specification/Infrastructure/ProductGrid/AddProductScorePropertySpec.php

```diff
@@ -0,0 +1,82 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Specification\Akeneo\Pim\Automation\DataQualityInsights\Infrastructure\ProductGrid;
+
+use Akeneo\Pim\Automation\DataQualityInsights\Domain\Model\ChannelLocaleRateCollection;
+use Akeneo\Pim\Automation\DataQualityInsights\Domain\Query\ProductEvaluation\GetLatestProductScoresQueryInterface;
+use Akeneo\Pim\Automation\DataQualityInsights\Domain\ValueObject\ChannelCode;
+use Akeneo\Pim\Automation\DataQualityInsights\Domain\ValueObject\LocaleCode;
+use Akeneo\Pim\Automation\DataQualityInsights\Domain\ValueObject\Rate;
+use Akeneo\Pim\Enrichment\Component\Product\Grid\Query\FetchProductAndProductModelRowsParameters;
+use Akeneo\Pim\Enrichment\Component\Product\Grid\ReadModel\Row;
+use Akeneo\Pim\Enrichment\Component\Product\Model\WriteValueCollection;
+use PhpSpec\Exception\Example\FailureException;
+use PhpSpec\ObjectBehavior;
+use Prophecy\Argument;
+
+class AddProductScorePropertySpec extends ObjectBehavior
+{
+    public function let(GetLatestProductScoresQueryInterface $getProductScores)
+    {
+        $this->beConstructedWith($getProductScores);
+    }
+
+    public function it_returns_no_rows_when_given_no_rows(FetchProductAndProductModelRowsParameters $queryParameters)
+    {
+        $this->add($queryParameters, [])->shouldReturn([]);
+    }
```

Comment:
As the `FetchProductAndProductModelRowsParameters` service is defined with `final` keyword it could not mocked like this way

```suggestion
    public function it_returns_no_rows_when_given_no_rows(ProductQueryBuilderInterface $productQueryBuilder)
    {
        $queryParameters = new FetchProductAndProductModelRowsParameters($productQueryBuilder, [], 'ecommerce', 'en_US');
        $this->add($queryParameters, [])->shouldReturn([]);
    }
```

Reply:
Done